### PR TITLE
PXC-3514: Merge galera 4.6

### PR DESCRIPTION
**Pull request by dutow**

### CODE REVIEW

File: gcache/src/GCache_seqno.cpp [hunk 1]
```diff
@@ -171,8 +171,14 @@ namespace gcache
 
         bool   loop(seqno >= seqno_released);
 
+        bool sleep_a_bit = false;
```
Comment (kamil-holubicki) [inline gcache/src/GCache_seqno.cpp:177]: Was removing of condition the good move? We ended up here with ugly sleep. Is it what we want?
On the other hand, I understand the maintainability argument, to be as close as possible to upstream. In such a case please add the description why we have this sleep solution here.

**Review by kamil-holubicki — APPROVED** [review]
Approved with 1 minor comment.

File: gcache/src/GCache_seqno.cpp (hunk 1 above)
Comment (dutow) [reply gcache/src/GCache_seqno.cpp:177]: I just checked this, and upstream fixed the issue differently (by just returning an error in this case) in 4.7. I'll try to see if that solution would work for us, but I'm not 100% sure.

For removing condition: this was a refactoring, removing the condition variable was just one part of it, I don't think keeping the old code would be a good idea.

Comment (dutow) [reply gcache/src/GCache_seqno.cpp:177]: I'm still getting failures when modifying this logic to what upstream does, I think we'll also need 1 or 2 other changes from 4.7. Let's keep this logic for now, and I'll see if I can remove it when we merge 4.7.

**Pull request merged**